blocks/rich-text/index.js

```diff
@@ -692,12 +667,22 @@ export class RichText extends Component {
 		} );
 	}
 
-	setContent( content = '' ) {
-		this.editor.setContent( renderToString( content ) );
+	setContent( content ) {
+		const { format } = this.props;
+		this.editor.setContent( valueToString( content, format ) );
 	}
 
 	getContent() {
-		return nodeListToReact( this.editor.getBody().childNodes || [], createTinyMCEElement );
+		const { format } = this.props;
+
+		switch ( format ) {
+			case 'string':
+				return this.editor.getContent();
+			default:
+				return this.editor.dom.isEmpty( this.editor.getBody() ) ?
```

Comment:
Aside: Is the `isEmpty` check here meant to be an optimization for the easy case of generating an empty string? Is `isEmpty` itself actually performant? Wondering if we're actually shooting ourselves in the foot with this for the more common case that `RichText` is non-empty.

Reply:
It's not an optimization, it's to ensure the paragraph block is considered empty consistently. If we remove it we can end up with content like `['']` which will break the default appender.

Reply:
I get the sense this has been discussed before, but if we have control over the elements we're producing, why are we keeping the empty string ?

Reply:
I can't tell honestly :) maybe @iseulde can tell more here.

Reply:
> but if we have control over the elements we're producing

Do we have control over that? This one is getting the nodes from a contenteditable field which can contain all sorts combination. I've seen `<p></p>`,`<p>''</p>`, `<p><br></p>`, `<p><br>''</p>`, `<p><strong></strong></p>` etc., all of which are empty.